core/templates/pages/exploration-editor-page/feedback-tab/services/thread-data-backend-api.service.ts

```diff
@@ -253,8 +253,19 @@ export class ThreadDataBackendApiService {
       text: newMessage
     }).toPromise().then((response: ThreadMessages) => {
       if (updatedStatus) {
-        this.openThreadsCount += (
-          (newStatus === ExplorationEditorPageConstants.STATUS_OPEN) ? 1 : -1);
+        let key: number = 0;
+        if (newStatus === ExplorationEditorPageConstants.STATUS_OPEN) {
```

Comment:
Can you explain this change?

Reply:
@nithusha21  basically, the problem was that it showed negative number as thread count, so to solve this whenever the status will be open we will increase the threadCount by one and for all other cases like fixed, ignored, compliment and not actionable we decrease it by one. But suppose we have 3 openthread and then we set a particular thread to ignored the count will decrease by one and again if we set the same thread to fixed, again the count decreases by one for that thread and thus go on decreasing. But actually for a single thread the count should decrease by one only and once only for all cases other than open. so to display the correct count of open threads I tried do it in such a way so that if the thread is anything other then open it will decrease the count by one only and for open increase by one. Please try to test it putting breakpoints then maybe you can get what I meant to say and tried to do and also please correct me if I am wrong at any point.

Reply:
I think this logic can be implemented in a simpler fashion right? Basically this is what you are trying to do. You know the number of threads before this status change. Now if the newStatus is open, then we need to increase the count. If it is not, but the old status is open, then we know that one thread closed, and hence we reduce the count. The count cannot possibly go below zero with this logic. In the existing code, the count was reduced if newStatus isn't open which is wrong. We need to make sure that the old status is open before reducing. I don't see why we need the 'key' variable you have here. 

Let me know if this isn't clear to you.

Reply:
Done, what you said is more simple.